SCClassLibrary/Common/Streams/FilterPatterns.sc

```diff
@@ -433,7 +434,9 @@ Pfindur : FilterPattern {
 			if (nextElapsed.roundUp(tolerance) >= localdur) {
 				// must always copy an event before altering it.
 				// fix delta time and yield to play the event.
-				inevent = inevent.copy.put(\delta, localdur - elapsed).yield;
+				remaining = localdur - elapsed;
+				if(inevent.isRest) { remaining = Rest(remaining) };
```

Comment:
IIUC, with this current version, if `delta` isn't a rest, and `inevent.isRest` is true for another reason, outgoing event's `delta` is still a rest (i don't know if this is okay or not but it feels a bit messy). would it be better here to check `delta.isRest`? that would also be more efficient -- for any event that isn't a rest, with the line above we'll cycle through all of the key/value pairs every time.

am i right in thinking that `inevent.isRest` is true and `delta` isn't a rest, and we make a copy of `inevent` with a different `delta`, then `inevent` should still be a rest?

Reply:
> would it be better here to check delta.isRest?

The variable `delta` is always a number, because the event-delta primitive unwraps Rests and rejects non-numeric types. So `delta.isRest` is always false.

I could change it to `inevent[\delta].isRest`.

> outgoing event's delta is still a rest (i don't know if this is okay or not but it feels a bit messy.

That's true. I didn't think it was especially harmful because `(degree: Rest(0), delta: 1)` and `(degree: Rest(0), delta: Rest(1))` are both rests, so there's no change in outward behavior. But we don't know what the user is doing with `anEvent[\delta]` so you're right that it would be better to make it more precise.

> am i right in thinking that inevent.isRest is true and delta isn't a rest, and we make a copy of inevent with a different delta, then inevent should still be a rest?

Yes. The bug occurs here if `delta` is the only Rest inside the event.

(Btw, aside, this is one of those places in the pattern library where the meanings of `event` and `inevent` are reversed. `inevent` should come from the argument or the previous yield and `event` should come from the stream. I never understood how this inverted convention got started but this isn't the only `embedInStream` method that makes this mistake.)

Reply:
> The variable delta is always a number, because the event-delta primitive unwraps Rests and rejects non-numeric types. So delta.isRest is always false.
> I could change it to inevent[\delta].isRest.

ah okay, i didn't realize `Event:-delta` is not just a member access, but that makes sense. yeah, i think limiting it that way would be better.

> (Btw, aside, this is one of those places in the pattern library where the meanings of event and inevent are reversed. inevent should come from the argument or the previous yield and event should come from the stream. I never understood how this inverted convention got started but this isn't the only embedInStream method that makes this mistake.)

possibly someone misinterpreting `inevent` as meaning 'the event we are in' rather than 'the input/incoming event' (is that the intended meaning?).

Reply:
OK, done.